src/Web.cpp

```diff
@@ -1685,34 +1685,81 @@ bool DumpNvsToJSONCallback(const char *key, void *data) {
 	return tagIdToJSON(key, obj);
 }
 
+// callback for writing a NVS entry to list
+bool DumpNvsToArrayCallback(const char *key, void *data) {
+	std::vector<String> *keys = (std::vector<String> *) data;
+	keys->push_back(key);
+	return true;
+}
+
+static String tagIdToJsonStr(const char *key) {
+	StaticJsonDocument<512> doc;
+	JsonObject entry = doc.createNestedObject(key);
+	if (!tagIdToJSON(key, entry)) {
+		return "";
+	}
+	String serializedJsonString;
+	serializeJson(entry, serializedJsonString);
+	return serializedJsonString;
+}
+
+uint16_t nvsIndex;
+
 static void handleGetRFIDRequest(AsyncWebServerRequest *request) {
-	String tagId = "";
+
+	String tagId, json = "";
+
 	if (request->hasParam("id")) {
 		tagId = request->getParam("id")->value();
 	}
-	if (tagId == "") {
-		// return all RFID assignments
-		AsyncJsonResponse *response = new AsyncJsonResponse(true, 8192);
-		JsonArray Arr = response->getRoot();
-		if (listNVSKeys("rfidTags", &Arr, DumpNvsToJSONCallback)) {
-			response->setLength();
-			request->send(response);
-		} else {
-			request->send(500, "error reading entries from NVS");
-		}
-	} else {
-		if (gPrefsRfid.isKey(tagId.c_str())) {
-			// return single RFID assignment
-			AsyncJsonResponse *response = new AsyncJsonResponse(false);
-			JsonObject obj = response->getRoot();
-			tagIdToJSON(tagId, obj);
-			response->setLength();
-			request->send(response);
-		} else {
-			// RFID assignment not found
-			request->send(404);
-		}
+
+	if ((tagId != "") && gPrefsRfid.isKey(tagId.c_str())) {
+		// return single RFID entry
+		json = tagIdToJsonStr(tagId.c_str());
+		request->send(200, "application/json", json);
+		return;
 	}
+	static std::vector<String> nvsKeys {};
+
+	nvsKeys.clear();
+	// Dumps all RFID-keys from NVS into key array
+	listNVSKeys("rfidTags", &nvsKeys, DumpNvsToArrayCallback);
+	if (nvsKeys.size() == 0) {
+		// no entries
+		request->send(200, "application/json", "[]");
+		return;
+	}
+	// construct chunked repsonse
+	nvsIndex = 0;
+	AsyncWebServerResponse *response = request->beginChunkedResponse("application/json",
+		[](uint8_t *buffer, size_t maxLen, size_t index) {
```

Comment:
If `nvsKeys` is not static any more, we have to capture the variable here so it's available in the lambda. To prevent a copy of the vector due to capture by value, we call std::move to transfer the variable into the lambda (capture by reference would result in undefined behaviour since the local variable will not exists any more when we are called).

```suggestion
		[nvsKeys = std::move(nvsKeys)](uint8_t *buffer, size_t maxLen, size_t index) {
```